src/Host/Client/Impl/Host/BrokerClient.cs

```diff
@@ -133,8 +139,8 @@ internal abstract class BrokerClient : IBrokerClient {
                 } catch (UnauthorizedAccessException) {
                     isValidCredentials = false;
                     continue;
-                } catch (HttpRequestException ex) {
-                    throw new RHostDisconnectedException("HTTP error while creating session: " + ex.Message, ex);
+                } catch (BrokerApiErrorException apiex) {
+                    throw new RHostDisconnectedException(apiex.ApiError);
```

Comment:
Might as well remove the status code from `RHostDisconnectedException` (since in many cases it doesn't have one), and just use `apiex` as the inner exception, I think. If anyone actually needs to find out why disconnect happened, they'll need to inspect `InnerException` in most cases anyway.